File: DONGUK/프로그래머스_WEEK_13/최고의 집합.py
def solution(n, s):
    answer = []
    
    q = s // n
    r = s % n
    
    if q == 0:
        answer.append(-1)
    else:
        for _ in range(n-r):
            answer.append(q)
            
        for _ in range(r):
            answer.append(q+1)
        
    
    return answer

File: JINSEOP/프로그래머스_WEEK13/최고의 집합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int s)
{
    vector<int> answer;

    if (n > s) {
        answer.push_back(-1);
        return answer;
    }

    for(int i = 1; i <= n; i++)
        answer.push_back(s / n);
    
    for(int i = 1; i <= s % n; i++)
        answer[n - i]++;

    return answer;
}

// 최대한 균등할 때, 곱이 최대가 된다!
// 문제가 어이가 없네..
